crates/benches/src/katana_bench/mod.rs

```diff
@@ -0,0 +1,74 @@
+#[cfg(test)]
+mod timings_stats;
+
+use std::time::Duration;
+
+pub const ENOUGH_GAS: &str = "0x100000000000000000";
+pub const BLOCK_TIME: Duration = Duration::from_secs(3);
+pub const N_TRANSACTIONS: usize = 2000;
+
+#[cfg(test)]
+mod tests {
+
+    use futures::future::join_all;
+    use starknet::accounts::Account;
+    use starknet::core::types::FieldElement;
+    use tokio::time::{sleep, Instant};
+
+    use super::*;
+    use crate::katana_bench::timings_stats::timetable_stats;
+    use crate::*;
+
+    #[tokio::test]
+    #[ignore] // needs a running katana --max-connections 10000 -b 3000
+    async fn bench_katana() {
+        let account_manager = account_manager().await;
+        let max_fee = FieldElement::from_hex_be(ENOUGH_GAS).unwrap();
+        let calldata_spawn = parse_calls(vec![BenchCall("spawn", vec![])]);
+        let calldata_move =
+            parse_calls(vec![BenchCall("move", vec![FieldElement::from_hex_be("0x3").unwrap()])]);
+
+        // generating all needed accounds
+        let accounts = join_all((0..N_TRANSACTIONS).map(|_| account_manager.next())).await;
+        let (spawn_txs, move_txs): (Vec<_>, Vec<_>) = accounts
+            .iter()
+            .map(|(account, nonce)| {
+                let spawn_call =
+                    account.execute(calldata_spawn.clone()).nonce(*nonce).max_fee(max_fee);
+                let move_call = account
+                    .execute(calldata_move.clone())
+                    .nonce(*nonce + 1u8.into())
+                    .max_fee(max_fee);
+                (spawn_call, move_call)
+            })
+            // .collect::<Vec<_>>();
+            .unzip();
+
+        // running a spawn for each account
+        join_all(spawn_txs.iter().map(|t| t.send())).await;
+        sleep(BLOCK_TIME).await;
```

Comment:
Katana will have longer block times in the future, potentially even a single block per instantiation. I'm curious how we can decouple benchmarking from the BLOCK_TIME? One idea is we incrementally ramp up the load testing and measure the transaction return latency? Since in Katana, iirc, executions are done synchronously. There isn't a mempool.

So the chart would basically be tx/s vs rtt, where rtt will start to increase as the sequencers queue get backlogged?

Reply:
I think the time slept is arbitrary here. I chose the time as block time to be sure it is cooled down before the main benchmark. 
I will be sure to add the rx/s - rtt measurement in the next PRs.